docs/manual/src/interface/functions.md

```diff
@@ -0,0 +1,70 @@
+# Functions
+
+The interface definition can contain public functions using the standard Rust syntax:
+
+```rust
+#[uniffi::declare_interface]
+mod example {
+    fn hello_world() -> String {
+        "Hello World!".to_owned()
+    }
+}
+```
+
+The arguments and return types must be types that are understood by UniFFI.
+
+UniFFI does not understand path references or type aliases, so things like
+the following will produce a compile-time error:
+
+```rust
+
+struct NotPartOfTheInterface {
+    value: u32,
+}
+
+#[uniffi::declare_interface]
+mod example {
+
+    type MyString = String
+
+    // Error: UniFFI doesn't know what "MyString" is.
+    fn hello_world() -> MyString {
+        "Hello World!".to_owned()
+    }
+
+    // Error: UniFFI doesn't know about the `NotPartOfTheInterface` type.
+    fn example(v: NotPartOfTheInterface) {
+        println!("value: {}", v.value);
+    }
+
+    // Error: UniFFI doesn't understand the `std::vec::` path; just use `Vec<T>`.
+    fn smallest(values: std::vec::Vec<i32>) -> i32 {
+        values.iter().min()
+    }
+}
+```
+
+## Optional arguments & default values
+
+Function arguments can be marked as optional with a default value specified.
+TODO: what will the Rust syntax for this be?
+
+The Rust code will declare this using a macro annotation:
+
+```rust
+#[uniffi::defaults(name="World")] // TODO: not even sure what syntax is possible here...
+fn hello_name(name: String) -> String {
```

Comment:
Probably not a great suggestion, but we might consider dropping this as a feature? Even in today's UDL I get a bit confused with the `optional type?` thing and it never does quite what I expect on the rust side. For example, let's say a new binding comes along which doesn't support this (eg, even the "other side" of Rust), it would leave the authors in a bit of a bind.

Reply:
Dropping the feature of specifying default values?
Glean would be very sad about that, we rely on default values in our Kotlin/Swift/Python APIs.
Rust already doesn't have default values and that's a bit annoying.
It's worth making it very clear that this might not be supported everywhere.

Reply:
Fair enough - I agree they are useful in languages that support them, and I have no interest in making glean sad, so 👍 . I'll leave this conversation unresolved for now so Ryan can decide whether to add a note or not about this.